src/main/java/seedu/mark/logic/commands/FavoriteCommand.java

```diff
@@ -0,0 +1,66 @@
+package seedu.mark.logic.commands;
+
+import static seedu.mark.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.List;
+
+import seedu.mark.commons.core.Messages;
+import seedu.mark.commons.core.index.Index;
+import seedu.mark.logic.commands.exceptions.CommandException;
+import seedu.mark.logic.commands.results.CommandResult;
+import seedu.mark.model.Model;
+import seedu.mark.model.bookmark.Bookmark;
+import seedu.mark.model.tag.Tag;
+import seedu.mark.storage.Storage;
+
+/**
+ * Favorites a bookmark identified using its displayed index from Mark.
+ */
+public class FavoriteCommand extends Command {
+    public static final String COMMAND_WORD = "favorite";
+    public static final String COMMAND_ALIAS = "fav";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Favorites the bookmark identified by the index number used in the displayed bookmark list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_FAVORITE_BOOKMARK_SUCCESS = "Add to my favorites: %1$s";
+    public static final String MESSAGE_FAVORITE_BOOKMARK_DUPLICATE = "This bookmark already exists in your favorites";
```

Comment:
To match the success message
```suggestion
    public static final String MESSAGE_FAVORITE_BOOKMARK_DUPLICATE = "This bookmark already exists in Favorites";
```